plugins/modules/site_workflow_manager.py

```diff
@@ -1407,140 +1594,183 @@ def get_diff_merged(self, config):
+                                            floor_name = site.get("name")
+                                            self.log("Floor '{}' has been created successfully.".format(floor_name), "INFO")
+
+                                            upload_path = site.get("upload_floor_image_path", None)
+                                            if upload_path:
+                                                self.log("Upload path found for floor '{}'. Starting upload floor map from '{}.'".
+                                                         format(floor_name, upload_path), "INFO")
+
+                                                map_details, map_status, success_message = self.upload_floor_image(site)
+                                                if map_details:
+                                                    self.log("Floor map for '{}' uploaded successfully: {}".
+                                                             format(floor_name, success_message), "INFO")
+                                                else:
+                                                    self.log("Floor map upload failed for '{}'. Please check the upload path and retry.".
+                                                             format(floor_name), "ERROR")
+                                            else:
+                                                self.log("No upload path provided for '{}'. Floor created without floor map.".
+                                                         format(floor_name), "INFO")
+                                else:
+                                    self.log("No valid task ID received from the 'creating_bulk_site' response.", "WARNING")
+                                    return None
+                            else:
+                                self.log("No response received from the 'creating_bulk_site' API call.", "WARNING")
+                                return None
+
+                task_detail_list = []
+                for each_config in self.have:
+                    site_name_hierarchy = each_config.get("site_name_hierarchy")
+
+                    if each_config.get("site_exists"):
+                        payload_new = self.change_payload_data(each_config.get("want"))
+                        if payload_new.get("type") == "area":
+                            self.msg = "Site - {0} does not need any update".format(site_name_hierarchy)
+                            self.log(self.msg, "INFO")
+                            self.update_not_needed_sites.append(payload_new.get("type") + ": " + site_name_hierarchy)
+                        elif payload_new.get("type") in ("building", "floor"):
+                            site_params = each_config.get("site_params")
+                            site_params["site_id"] = each_config.get("site_id")
+                            site_type = site_params.get("type")
+
+                            if self.site_requires_update(each_config):
```

Comment:
self.log("Site requires update, starting update for type: {}".format(site_type), "DEBUG")

Reply:
Added